src/main/scala/io/iohk/ethereum/blockchain/sync/regular/ChainManagerService.scala

```diff
@@ -0,0 +1,28 @@
+package io.iohk.ethereum.blockchain.sync.regular
+
+import cats.data.NonEmptyList
+import io.iohk.ethereum.blockchain.sync.regular.BlockImporter.BlockImportType
+import io.iohk.ethereum.domain.Block
+import io.iohk.ethereum.network.Peer
+import monix.eval.Task
+
+//not used atm, a part of the future ExecutionSync
+object ChainManagerService {
+
+  type ValidationError
+  type Branch
+
+  //saves blocks to the storage
+  def processBlocksFromFetcher(peer: Peer): Task[Either[ValidationError, Branch]] = ???
+
+  //saves blocks to the storage
+  def processMinedBlocks(): Task[Either[ValidationError, Branch]] = ???
```

Comment:
what is the reason to process a fetched and a mined block differently ?

Reply:
they are coming from different paths: fetched blocks are coming from the BlockFetcher, mined blocks are coming from the controller and mined blocks can't fit the queue of blocks formed by a BlockFetcher that was fetching an explicit range